Runar/Runar/Services/Persistence layer/CoreDataService.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  CoreDataService.swift
+//  Runar
+//
+//  Created by Andrei on 11.02.2023.
+//
+
+import UIKit
+import CoreData
+
+protocol PesistenceService {
+    var runes: [RuneModel] { get }
+    
+    func addUpdateRune(
+        _ rune: RuneModel,
+        completionHandler: @escaping ((Result<Void, Error>) -> Void)
+    )
+}
+
+final class CoreDataService: PesistenceService {
+    private let context: NSManagedObjectContext? = {
+        let context = (UIApplication.shared.delegate as? AppDelegate)?.persistentContainer?.viewContext
```

Comment:
Could, you, please, explain, why do you use PesistenceService like a parent and retrieve viewContext from AppDelegate?

Reply:
-PesistenceService - interface, for using eventualy in high level abstraction 
-AppDelegate store container for all App, context we use when we want to work with database, and ask the container to give us context (we could work not with one context as well)